### PR TITLE
api: implement a `next` page after authentification

### PR DESCRIPTION
**Pull request by privat**
TODO: The UI part (www/javascripts/ng-app.js & www/directives/player/menu.html)

### CODE REVIEW

File: src/api/api_auth_shibuqam.nit [hunk 1]
```diff
@@ -94,7 +99,10 @@ class ShibCallback
 			config.players.save player
 		end
 		session.player = player
-		res.redirect "/player"
+
+		# Redirect
+		var page = session.auth_next or else "/player"
```
Comment (Morriar) [inline src/api/api_auth_shibuqam.nit]: Maybe this should be pulled up in the common ancestor of Github and Shibuqam login?

Comment (privat) [reply src/api/api_auth_shibuqam.nit]: What about adding common services in `APIHandler` ?

Comment (Morriar) [reply src/api/api_auth_shibuqam.nit]: The next page one? Maybe, do you have another example than login/logout where the `next` information is useful?

I can see the mission one but I don't like the semantics of passing the next uri in one case and guessing it in another case.

Following the current popcorn philosophy, this should be in the AuthHandler.

Comment (privat) [reply src/api/api_auth_shibuqam.nit]: AuthHadler is not documented, I not sure what is its role but it seems unrelated to the classes GithubLogin, GithubCallback, ShibLogin and ShibCallback.

Comment (Morriar) [reply src/api/api_auth_shibuqam.nit]: My bad, I though it was this one. It appears that ShibUqam and Github does not have a common ancestor more specific than APIHandler...

But I don't like the id of putting everything in APIHandler, it makes it API dirtier :/

Comment (privat) [reply src/api/api_auth_shibuqam.nit]: If I introduce a common abstract superclass in api_auth with the common services/code, is that OK for you?

Comment (privat) [reply src/api/api_auth_shibuqam.nit]: Eg `abstract class AuthLogin`

Comment (Morriar) [reply src/api/api_auth_shibuqam.nit]: Perfect

**Comment by privat** [comment]
reroll with a prior factorization of services common to github and shib

**Comment by ppepos** [comment]
Noice! +1

File: src/api/api_auth.nit [hunk 2]
```diff
@@ -57,6 +60,48 @@ class Logout
 	end
 end
 
+# Common services related to authentication/login.
+abstract class AuthLogin
+	super APIHandler
+
+	# Extract a possible next page from the GET arguments and store it in the session for latter
```
Comment (lbajolet) [inline src/api/api_auth.nit:67]: :s/latter/later

**Comment by lbajolet** [comment]
+1

**Pull request merged**